### PR TITLE
report: more attractive table/URL rendering

### PR DESCRIPTION
**Pull request by paulirish**
### Before and after
image

1. you can see WAY more of the URL because good CSS.
2. meanwhile we're more aggressively eliding non-readable strings in the URL

partial fix for #3827 but its not totally complete.

### CODE REVIEW

File: lighthouse-core/report/v2/renderer/details-renderer.js [hunk 1]
```diff
@@ -66,23 +66,30 @@ class DetailsRenderer {
...
     }
 
-    const element = this._renderText({
-      type: 'url',
-      text: displayedURL,
+    const element = this._dom.createElement('div', 'lh-text__url');
+    element.appendChild(this._renderText({
+      text: displayedPath,
+    }));
+    const hostElem = this._renderText({
```
Comment (patrickhulce) [inline lighthouse-core/report/v2/renderer/details-renderer.js]: does this need to be conditional on `displayedHost`?

File: lighthouse-core/report/v2/renderer/util.js [hunk 2]
```diff
@@ -147,8 +147,15 @@ class Util {
     }
 
     const MAX_LENGTH = 64;
-    // Always elide hash
+    // Always elide git hash
```
Comment (patrickhulce) [inline lighthouse-core/report/v2/renderer/util.js]: how about just `hex hash` :)

File: lighthouse-core/report/v2/renderer/util.js [hunk 3]
```diff
@@ -147,8 +147,15 @@ class Util {
...
     name = name.replace(/([a-f0-9]{7})[a-f0-9]{13}[a-f0-9]*/g, `$1${ELLIPSIS}`);
+    // Also elide other hash-like mixed-case strings // lol this is crazy
+    name = name.replace(/([a-zA-Z0-9-_]{9})(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])[a-zA-Z0-9-_]{10,}/g,
+        `$1${ELLIPSIS}`);
+    // Also elide long number sequences
+    name = name.replace(/(\d{3})\d{6,}/g, `$1${ELLIPSIS}`);
+    // Merge any adjacent ellipses
+    name = name.replace(/\u2026+/g, '\u2026');
```
Comment (patrickhulce) [inline lighthouse-core/report/v2/renderer/util.js]: `\u2026` is ellipsis right, so just pass `ELLIPSIS` here like the others? :)

**Comment by paulirish** [comment]
> do you know how this impacts cells that are in the url column but are actually other types like code? i'm thinking of the css usage audit in particular

good call!! it wasn't ideal. but after some fixes:

image
image
image


updated.

**Comment by paulirish** [comment]
_awaiting CI results. when green, land away._

**Comment by patrickhulce** [comment]
geez since when did "Unable to connect to Chrome" start failing 100% of the time??

**Pull request merged**